lib/core/request.c

```diff
@@ -357,8 +357,10 @@ int h2o_req_validate_pseudo_headers(h2o_req_t *req)
 
 h2o_handler_t *h2o_get_first_handler(h2o_req_t *req)
 {
-    h2o_hostconf_t *hostconf = h2o_req_setup(req);
-    setup_pathconf(req, hostconf);
+    if (req->pathconf == NULL) {
+        h2o_hostconf_t *hostconf = h2o_req_setup(req);
+        setup_pathconf(req, hostconf);
+    }
```

Comment:
This looks like a change unrelated to this PR. If it is necessary, could you clarify why we want to make this change?

Reply:
It's related. As of this PR, `h2o_get_first_handler` can be called twice (one by h2o_req_can_stream_request, another by h2o_req_should_forward_expect). Unless we add this change, h2o_req_setup and setup_pathconf will be called twice (I don't think it causes any serious failures but it's just wasteful)